src/storage/store/rel_table.cpp

```diff
@@ -156,40 +124,38 @@ void DirectedRelTableData::scanLists(Transaction* transaction, RelTableScanState
 void DirectedRelTableData::insertRel(const shared_ptr<ValueVector>& boundVector,
     const shared_ptr<ValueVector>& nbrVector,
     const vector<shared_ptr<ValueVector>>& relPropertyVectors) {
-    auto boundTableID =
-        boundVector->getValue<nodeID_t>(boundVector->state->selVector->selectedPositions[0])
-            .tableID;
-    if (!adjColumns.contains(boundTableID)) {
+    if (!isSingleMultiplicity()) {
         return;
     }
-    auto adjColumn = adjColumns.at(boundTableID).get();
     auto nodeOffset =
         boundVector->readNodeOffset(boundVector->state->selVector->selectedPositions[0]);
     // TODO(Guodong): We should pass a write transaction pointer down.
     if (!adjColumn->isNull(nodeOffset, Transaction::getDummyWriteTrx().get())) {
         throw RuntimeException(
             StringUtils::string_format("Node(nodeOffset: %d, tableID: %d) in RelTable %d cannot "
                                        "have more than one neighbour in the %s direction.",
-                nodeOffset, boundTableID, tableID, getRelDirectionAsString(direction).c_str()));
+                nodeOffset,
+                boundVector->getValue<nodeID_t>(boundVector->state->selVector->selectedPositions[0])
+                    .tableID,
+                tableID, getRelDirectionAsString(direction).c_str()));
     }
     adjColumn->writeValues(boundVector, nbrVector);
     for (auto i = 0u; i < relPropertyVectors.size(); i++) {
-        auto propertyColumn = getPropertyColumn(boundTableID, i);
+        auto propertyColumn = getPropertyColumn(i);
         propertyColumn->writeValues(boundVector, relPropertyVectors[i]);
     }
 }
 
 void DirectedRelTableData::deleteRel(const shared_ptr<ValueVector>& boundVector) {
     auto boundNode =
         boundVector->getValue<nodeID_t>(boundVector->state->selVector->selectedPositions[0]);
-    if (!adjColumns.contains(boundNode.tableID)) {
+    if (!isSingleMultiplicity()) {
```

Comment:
this is not related to this PR. But should we error in this case instead of return? Ditto for insert and update

Reply:
No, the operator doesn't know whether the edge is stored as list or column, it needs to check whether we should delete from column